relayer/client-tx.go

```diff
@@ -273,7 +272,7 @@ func FindMatchingClient(source, counterparty *Chain, clientState *ibctmtypes.Cli
 		}
 
 		// check if the client states match
-		if IsMatchingClient(*clientState, *existingClientState) {
+		if IsMatchingClient(*clientState, *existingClientState) && !existingClientState.IsFrozen() {
```

Comment:
I don't get why this wasn't caught on line 337.

`clientState` is not frozen but `existingClientState` is. So wouldn't it have two fields `IsFrozen` and `FrozenHeight` that are different?

Reply:
I added this as a sanity check. In the issues reported, the client isn't frozen but expired. Expired clients don't have frozen heights